app/services/statistics/base.rb

```diff
@@ -19,7 +21,7 @@ def id
     end
 
     def url
-      @url || "statistics/#{id}"
+      @url || build_url("statistics/#{id}", @organization_id, @location_id)
```

Comment:
Why are we building the url again here? shouldn't it already be set by the initializer?

Reply:
The problem is when you pass blank url to initializer, in that case the build_url returns nil, I think it hit me in tests. I think we rarely set some explicit url and just build it based on id when `url` method is called but we should keep supporting custom URLs.

Reply:
this would be resolved in the second commit since the url would become mandatory